### PR TITLE
glew: default enableEGL to true

### PR DESCRIPTION
**Pull request by flokli**
## Description of changes
re-opened https://github.com/NixOS/nixpkgs/pull/328805.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (cross from x86_64-linux)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/glew/default.nix
Comment (SomeoneSerge) [inline pkgs/development/libraries/glew/default.nix:11]: Note: parens are redundant, and the bulk of Nixpkgs switched to nixmft-rfc-style by now

**Review by SomeoneSerge — APPROVED** [review]

File: pkgs/development/libraries/glew/default.nix
Comment (flokli) [reply pkgs/development/libraries/glew/default.nix:11]: I'll leave that up to the big reformatting :-)

Comment (SomeoneSerge) [reply pkgs/development/libraries/glew/default.nix:11]: Yeah-yeah, let's keep adding noise to the git-blames

**Comment by K900** [comment]
This regressed mangoapp, and by extension gamescope-session/Jovian. `glewInit()` simply fails.

**Comment by K900** [comment]
Also, I'm sorry if this comes off as harsh, but like, why? GLEW is ancient, and extremely finicky, and this was just ... merged, without even a rationale?

**Comment by SomeoneSerge** [comment]
(replying because I posted an approval)

> GLEW is ancient, and extremely finicky,

IMO not a very good reason to fear touching things or to keep surprising and undocumented defaults. Maybe now we can at least add a warning comment

>  regressed mangoapp, and by extension gamescope-session/Jovian. glewInit() simply fails.

It's sad there wasn't any passthru tests to show these before merging. I wonder if it's something we could test within a sandbox or something hardware-specific

> why?

There was an app which used GLEW and EGL. Attempting to update the default seems like a reasonable upkeep move

> Needs more investigation either way.

👍🏻 and thanks for promptly reverting

**Comment by K900** [comment]
I'm not saying we should fear touching things, I'm just saying we should be really careful with things like this, and we should definitely document why we're making changes when making them. Looking into it more, it seems like enabling EGL just completely disables GLX code paths, which breaks things. The Arch people have a patch for this we may want to steal: https://gitlab.archlinux.org/archlinux/packaging/packages/glew/-/blob/main/egl+glx.patch?ref_type=heads.

**Comment by flokli** [comment]
I re-rolled this, including the patch from archlinux, over at https://github.com/NixOS/nixpkgs/pull/333707.

**Pull request merged**
